test/tests/functional/pbs_ralter.py

```diff
@@ -968,15 +994,54 @@ def test_check_alternate_nodes_advance_resv(self):
         attrs = {'resv_nodes': (MATCH_RE, re.escape(free_node))}
         self.server.expect(RESV, attrs, id=rid1)
 
-    def test_check_alternate_nodes_standing_resv(self):
+    def test_check_alternate_nodes_advance_resv_starttime(self):
+        """
+        This test confirms that an advance reservation can be extended even if
+        there is a conflicting reservation but there are nodes in the complex
+        that satisfy the resource requirements of the reservation.
+
+        Two back to back advance reservations are submitted that use the
+        same nodes in the complex to test this and start time of the former
+        reservation is altered.
         """
-        This test confirms that a standing reservation's end time can be
-        moved back even if there is a conflicting reservation but there are
+        duration = 120
+        shift = 10
+        offset1 = 60
+        offset2 = 180
+
+        rid1, start1, end1 = self.submit_and_confirm_reservation(
+            offset2, duration, select="1:ncpus=4")
+
+        self.server.status(RESV, 'resv_nodes', id=rid1)
+        resv_node = self.server.reservations[rid1].get_vnodes()[0]
+
+        select = "1:vnode=" + resv_node + ":ncpus=4"
+        rid2, start2, end2 = self.submit_and_confirm_reservation(
+            offset1, duration, select=select)
+
+        attrs = {'resv_nodes': (MATCH_RE, re.escape(resv_node))}
+        self.server.expect(RESV, attrs, id=rid2)
+
+        attrs = {'resources_available.ncpus': (GT, 0)}
+        vnode_dict = self.server.filter(NODE, attrs)
+        vnode_list = vnode_dict.values()[0]
```

Comment:
I understand that you like having the natural vnode with 0 cpus, but if you used it you wouldn't have to do this.  This just adds to confusion to the test and takes extra test time.

Reply:
Ok, removed :)